### ISSUE TITLE
Swift Version

### ISSUE THREAD

**Issue by Rufy86** [John-Lluch/SWRevealViewController#724]
is it possible to have a swift version?

**Comment by iDevelopper** [comment]
What for? You can use this ObjC library in a Swift project...

**Comment by Rufy86** [comment]
Yes but this library don't work in Xcode 9. I've open and issue about this.

after that, every code is coded in swift by compiler. than, why to have again a library written in OBJ-C? therefore this library is not updated from 2 years

**Comment by iDevelopper** [comment]
I know but I am not the developer of this library. Why not working on Xcode 9? I have Xcode 8.3.3. What is the issue?

**Comment by Rufy86** [comment]
If I'm working on Xcode 9 is because I've reasons. if you aren't the developer of this library, please, don't intrude into the conversation. my issues are for developers of this library. I this is a great library and Whether this library is maintained and converted into swift can have a great potential for improvement

**Comment by iDevelopper** [comment]
I know perfectly this library and I helped a lot of developer. It is why I answer. It is allowed, right? I asked you what are the issues on Xcode 9, I not asked you why do you use Xcode 9...

**Comment by iDevelopper** [comment]
I also know the developer and I don't think there will be a swift version. You are welcome to fork it!

**Comment by Rufy86** [comment]
I'm sorry. I did not understand correctly.

I've open an issue about this problem, anyway they are these:
1)Console say me this:Unknown class SWRevealViewController in Interface Builder file.

2)if I connect button to swrevealviewcontroller action, the app crash with this error:Undefined symbols for architecture arm64: "OBJC_CLASS$_SWRevealViewController" referred from:

**Comment by iDevelopper** [comment]
Ok, I don't think it is relative to Xcode 9, but I will install it to test.

**Comment by Rufy86** [comment]
>  also know the developer and I don't think there will be a swift version

This is so sad.

> You are welcome to fork it!
I hope to use it again in objective-c because I'm not a genius of swift and unfortunately I've no time to try to make it in swift, but if I have no choice..........

> I don't think it is relative to Xcode 9
in Xcode 8.x.x i've no problem

**Comment by Rufy86** [comment]
anyway, thank you for your interest about this

**Comment by iDevelopper** [comment]
Downloading Xcode 9, but my connection is very slow. Do you drag the .h and .m in your project or do you use Cocoapods? (Cocoapods version is not up to date). The latest version is 2.4 and you can see it in the .h

**Comment by Rufy86** [comment]
I drag the .h and .m file. I don’t use cocoapod because I don’t like it

**Comment by iDevelopper** [comment]
Tested and works!

In your storyboard check it is like this screenshot:

![2017-08-16_07-01-22](https://user-images.githubusercontent.com/1782450/29348069-2e66462c-8251-11e7-911b-95f2433a9576.png)

And Product->Clean
And Alt+Product->Clean Build Forder...

**Comment by Rufy86** [comment]
I'm sorry but the problem still

this is the error:
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_SWRevealViewController", referenced from:
      objc-class-ref in MyClassViewController.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by iDevelopper** [comment]
Ok this is the 2) point. How did you connect your button? Can you elaborate? What is your code for that?

**Comment by Rufy86** [comment]
I connect the button by Storyboard @IBOutlet weak var menuButton: UIBarButtonItem!

this is the code that when there is, don't build.
but when I comment this if everything work

```
        if self.revealViewController() != nil {
            menuButton.target = self.revealViewController()
            menuButton.action = #selector(SWRevealViewController.revealToggle(_:))
            self.view.addGestureRecognizer(self.revealViewController().panGestureRecognizer())
        }
```

**Comment by iDevelopper** [comment]
Only link error? No compil errors?

**Comment by Rufy86** [comment]
That's right.

Now I've tried to open the project with Xcode 8. same problem. Now I'm trying to do the opposite. create it in Xcode 8 and open it with Xcode 9

**Comment by Rufy86** [comment]
If I create a the project with Xcode 8 and I open it with Xcode 9 work. I have also 45 warning but work. or build without error

**Comment by iDevelopper** [comment]
Did you try to delete the derived data folder?

**Comment by Rufy86** [comment]
No. How I do it?

**Comment by iDevelopper** [comment]
Go to Xcode->Preferences

Tab: Locations

Click on the arrow and in the finder delete all the folder for this application:

![2017-08-16_13-01-47](https://user-images.githubusercontent.com/1782450/29360447-36ea8268-8283-11e7-9e24-7051d5ef6313.png)

**Comment by Rufy86** [comment]
I've found the problem.
![problem](https://user-images.githubusercontent.com/17876640/29369627-7000e75c-82a3-11e7-9435-9b786d278494.jpg)

when you add the .h and .m file Though the target flag is checked, the flag don't be checked. when I've checked, The problem has disappeared.

however, a swift version I think be better!

anyway, thank you very much for your help

**Comment by Rufy86** [comment]
nothing.
the first problem still again

Unknown class _TtC38MyProjectName22SWRevealViewController in Interface Builder file.

**Comment by Rufy86** [comment]
Found it!
![29348069-2e66462c-8251-11e7-911b-95f2433a9576](https://user-images.githubusercontent.com/17876640/29371574-fdad5e5a-82a8-11e7-9fd9-f1925dab7422.png)

without check work perfectly

**Comment by iDevelopper** [comment]
Yes when you drag a .m .h into a project there is a dialog box where you can check the targets. Well done! For the Swift version, I do not think there will be one except a fork. However, I have a similar library (see my profile) and I have a Swift version (not yet on line) but soon. I can send you the files if you want.

**Comment by iDevelopper** [comment]
If you are interested, open an issue to https://github.com/iDevelopper/PBRevealViewController

**Comment by rajeevjaiswal** [comment]
Same issue. Any Solution?

**Comment by rajeevjaiswal** [comment]
Solved. viewcontroller.m file target membership was not checked
